src/GraphODataTemplateWriter/CodeHelpers/CSharp/TypeHelperCSharp.cs

```diff
@@ -431,5 +431,89 @@ public static string GetCoreLibraryType(this string @namespace, string type)
                 return $"Microsoft.Graph.{type}";
             }
         }
+
+        /// <summary>
+        /// Use this method to get the method information for all of the methods.
+        /// Used in *MethodRequestBuilder
+        /// </summary>
+        /// <param name="methods">A list of OdcmMethods.</param>
+        /// <param name="namespace">The namespace for the methods.</param>
+        /// <returns>An ordered (to minimize diffs caused by reorder in the CSDL) list of 
+        /// MethodInfo objects used to construct the request builders.</returns>
+        public static List<MethodInfo> GetAllMethodsInfo(this List<OdcmMethod> methods, string @namespace)
+        {
+            return methods.Select(m =>
+            {
+                var parameters = m.Parameters
+                    .Select(p =>
+                    {
+                        var type = p.Type.GetTypeString(@namespace);
+                        var name = p.Name.ToLowerFirstChar();
+                        var parameterName = p.Name.GetSanitizedParameterName();
+                        var propertyName = p.Name.ToCheckedCase();
+
+                        // Adds support for classes ending in "Request" that have been dismabiguated.
+                        if (type.EndsWith("Request"))
+                        {
+                            type = String.Concat(type, "Object");
+                        }
+
+                        // Adjust the type string
+                        if (p.IsCollection)
+                        {
+                            type = $"IEnumerable<{type}>";
+                        }
+                        else if (!p.Type.IsTypeNullable() && p.IsNullable)
+                        {
+                            type += "?";
+                        }
+
+                        return new ParameterInfo(type, name, parameterName, propertyName, p.IsNullable);
+                    })
+                    .OrderBy(p => p.IsNullable ? 1 : 0); // I suspect this could cause a problem if new overloads are added.
+
+                var paramStrings = parameters.Select(p => string.Format(",\n            {0} {1}", p.Type, p.ParameterName));
```

Comment:
nit: is there any specific reason of using string format instead of string interpolation when you don't have a repetition of parameters use?

Reply:
No reason, it existed this way before and I moved it unchanged. Making the change now. A bit cleaner to look at now.